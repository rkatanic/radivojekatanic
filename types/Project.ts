export interface ProjectType {
  slug: string;
  title: string;
  description: string;
  creationDate: string;
  content: string;
  updated: ProjectUpdated;
  links: ProjectLink[];
  icon?: string;
}

export interface ProjectUpdated {
  description: string;
  date: string;
}

export interface ProjectLink {
  title: string;
  href: string;
  type: ProjectLinkType;
}

export type ProjectLinkType = "application" | "image" | "readme";
