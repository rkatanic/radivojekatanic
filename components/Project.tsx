import { FiArrowRight } from "react-icons/fi";
import { DATE_FORMAT } from "utils/dateTimeUtils";
import { ProjectType } from "../types/Project";
import { TbSeeding } from "react-icons/tb";

interface Props {
  project: ProjectType;
}

const Project = ({
  project: { title, description, creationDate },
}: Props): JSX.Element => (
  <div className="flex gap-4 border-b border-slate-800 pb-8">
    <div className="flex h-14 w-14 items-center justify-center rounded-full border border-slate-800 bg-gradient-to-t from-slate-800/70 to-slate-800/10">
      <TbSeeding size="1.25rem" className="stroke-sky-500" />
    </div>
    <div className="flex-1">
      <p className="mb-1 text-sm text-slate-500">
        {DATE_FORMAT.format(new Date(creationDate))}
      </p>
      <h3 className="font-serif text-lg font-semibold tracking-wide text-slate-200">
        {title}
      </h3>
      <p className="mt-2">{description}</p>
      {/* <button className="border-tran mt-4 flex items-center gap-2 border-b border-transparent text-sm text-sky-400 hover:border-sky-500">
        View more <FiArrowRight />
      </button> */}
    </div>
  </div>
);

export default Project;
