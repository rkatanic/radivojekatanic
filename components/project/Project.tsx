import Link from "next/link";
import { FiChevronRight } from "react-icons/fi";
import { DATE_FORMAT } from "utils/dateTimeUtils";
import { ProjectType } from "../../types/Project";

interface Props {
  project: ProjectType;
  index: number;
}

const Project = ({
  project: { slug, title, description, creationDate },
  index,
}: Props): JSX.Element => (
  <Link className="flex w-full" href={`/projects/${slug}`}>
    <div className="group relative flex border border-gray-300 bg-slate-50 shadow-sm transition-[border] duration-200 hover:border-sky-400 dark:border-gray-700 dark:bg-gray-900 dark:shadow-md dark:hover:border-sky-400">
      <div className="flex-1 p-4">
        <div className="mb-4 text-xl font-medium tracking-wide dark:border-gray-700 dark:text-gray-200">
          {title}
        </div>
        <div className="absolute top-6 right-4 z-10 flex items-end gap-1 border border-gray-300 py-0.5 px-1 text-xs font-light tracking-wide text-gray-600 dark:border-gray-700 dark:text-gray-300">
          <span className="text-gray-400">#</span> 0{index + 1}
        </div>
        <div className="absolute bottom-6 right-4 z-10 flex items-end gap-1 text-sm font-light tracking-wide text-gray-400 group-hover:text-sky-500 dark:text-gray-500 dark:group-hover:text-sky-300">
          View <FiChevronRight className="mb-0.5" />
        </div>
        <p className="text-lg leading-relaxed tracking-wide text-gray-500 dark:text-gray-400">
          {description}
        </p>
        <hr className="my-4 dark:border-gray-800" />
        <div className="mb-2 flex items-center gap-2 text-sm text-gray-400 dark:text-gray-400">
          {DATE_FORMAT.format(new Date(creationDate))}
        </div>
      </div>
      <div className="xbg-gradient-to-r relative h-full w-44 overflow-hidden from-gray-900 to-gray-800/40">
        <div className="cube group-hover:animate-[rotate_30s_linear_infinite]">
          <div className="sphere-one inner"></div>
          <div className="sphere-two inner"></div>
          <div className="sphere-three inner"></div>
          <div className="sphere-four inner"></div>
          <div className="sphere-five inner"></div>
          <div className="sphere-six inner"></div>
        </div>
      </div>
    </div>
  </Link>
);

export default Project;
