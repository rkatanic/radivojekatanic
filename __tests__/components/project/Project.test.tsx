import Project from "@/components/project/Project";
import { render } from "@testing-library/react";

describe("Project", (): void => {
  it("should render", (): void => {
    const project = {
      slug: "project-slug",
      title: "Project Title",
      description: "Project description.",
      creationDate: "10-10-2010",
    };
    const { baseElement } = render(<Project project={project} />);

    expect(baseElement).toMatchSnapshot();
  });
});
