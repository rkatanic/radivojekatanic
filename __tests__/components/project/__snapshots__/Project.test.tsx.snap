// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should render 1`] = `
<body>
  <div>
    <div
      class="group relative flex border-2 border-gray-200 bg-slate-50 shadow-sm transition-[border] duration-200 hover:border-sky-400 dark:border-gray-800 dark:bg-gray-900 dark:shadow-md dark:hover:border-sky-400"
    >
      <div
        class="flex-1 p-4"
      >
        <div
          class="mb-4 text-2xl font-medium dark:border-gray-700 dark:text-gray-200"
        >
          Project Title
        </div>
        <div
          class="absolute top-4 right-4 z-10 flex items-end gap-1 border-2 border-gray-200 py-1 px-2 text-xs font-medium tracking-wide text-gray-600 dark:border-gray-800 dark:text-gray-300"
        >
          <span
            class="text-gray-400"
          >
            #
          </span>
           0
          NaN
        </div>
        <div
          class="group absolute bottom-6 right-4 z-10 flex items-center gap-2 text-xs font-medium uppercase tracking-wide text-gray-400 group-hover:text-sky-500 dark:text-gray-500 dark:group-hover:text-sky-300"
        >
          View
          <div
            class="h-[2px] w-3 bg-gray-400 dark:bg-gray-500 dark:group-hover:bg-sky-300"
          />
        </div>
        <p
          class="text-lg leading-relaxed tracking-wide text-gray-500 dark:text-gray-400"
        >
          Project description.
        </p>
        <hr
          class="my-4 dark:border-gray-800"
        />
        <div
          class="mb-2 flex items-center gap-2 text-sm uppercase tracking-wide text-gray-600 dark:text-gray-400"
        >
          Oct 10, 2010
        </div>
      </div>
      <div
        class="xbg-gradient-to-r relative h-full w-44 overflow-hidden from-gray-900 to-gray-800/40"
      >
        <div
          class="cube group-hover:animate-[rotate_30s_linear_infinite]"
        >
          <div
            class="sphere-one inner"
          />
          <div
            class="sphere-two inner"
          />
          <div
            class="sphere-three inner"
          />
          <div
            class="sphere-four inner"
          />
          <div
            class="sphere-five inner"
          />
          <div
            class="sphere-six inner"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
